boilerplate/frontend/supertokens-react-custom/src/hooks/useSessionInfo.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+import Session from "supertokens-web-js/recipe/session";
+
+function useSessionInfo() {
+    const [isUserLoggedIn, setIsUserLoggedIn] = useState<undefined | boolean>(undefined);
```

Comment:
```suggestion
    const [isUserLoggedIn, setIsUserLoggedIn] = useState<{isLoading: true} | {isLoading: false, sessionExists: boolean}>(undefined);
```

Reply:
And whenever you call this function, you have to check is loading first and render a spinner

Reply:
Changed the hook response and also I have added a spinner